### DISCUSSION TITLE
Why django-unicorn doesnt update hmtl-collection?

### DISCUSSION THREAD

**Discussion by OwnMath** [django-commons/django-unicorn#401] [Q&A] [answered]
There is **`django-unicorn 0.44.0`**'s component configuration.

`refresh.py`

    from django.db import connection
    from django_unicorn.components import UnicornView
    from datamarket.models import Clients
    
    
    class RefreshView(UnicornView):
        clients = None
        count = None
    
        def get_client(self):
            self.count = Clients.objects.all().count()
            self.clients = Clients.objects.all().order_by("surname")[:10]
    
        def az(self):
            self.clients = Clients.objects.all().order_by("surname")[:10]
    
        def za(self):
            self.clients = Clients.objects.all().order_by("-surname")[:10]
    
        def mount(self):
            self.clients = Clients.objects.all().order_by("surname")[:10]
            self.count = Clients.objects.all().count()

`refresh.html`

        <div>
             <button class="btn" unicorn:click="get_client()">Update</button>
             <button class="btn" unicorn:click="az()">A-Z</button>
             <button class="btn" unicorn:click="za()">Z-A</button>
             <p> Total {{ count }} records</p>
         </div>
    
       <table>
                    <thead>
                        <th>Surname</th>
                        <th>Name</th>
                        <th>Age</th>
                    </thead>
                    <tbody>
                        {% for c in clients %}
                        <tr>
                            <td>{{ c.surname }}</td>
                            <td>{{ c.name }} </td>
                            <td>{{ c.age }}</td>
                        </tr>
                        {% empty %}
                        <tr>
                            <td colspan="3">No found</td>
                        </tr>
                        {% endfor %}
                    </tbody>
    
       </table>

The `mount()` function doing well when I refresh page, it changes `clients` value in html. Also `count` value updating well too when I call `get_clients()` by button.
But it doesnt change client collection in table when I call `get_clients()`, `az()`, `za()` by button. Why?

It's worked literally week ago and now I don't get any errors and dont have a clue why it's broken.

**Comment by OwnMath** [comment, marked as answer]
Problem solved.

html-file need **one general `</div>`**

**Reply by adamghill** [reply]
I'm glad you figured out was going on! There is some code that tries to alert if there isn't a surrounding element, but it sounds like that wasn't happening for you. Sorry about that.
